Извините. Выполнила домашнее задание по уроку 5.

TnWqUe opened the pull request:
Извините. Выполнила дз по уроку 5.

TnWqUe committed:
Извините. Выполнила домашнее задание по уроку 5.

Diagorn commented on ru/turing/courses/lesson5/Main.java:
```diff
@@ -1,7 +1,67 @@
+        System.out.println("Пример сортировки пузырьком для массива int[]");
+        //объявляем и инициализируем массив
+        int[] intArray = {1, 5, 4, 3, 7};
+        System.out.println("Исходный массив: "+Arrays.toString(intArray));
```
```suggestion
        System.out.println("Исходный массив: " + Arrays.toString(intArray));
```
Ctrl + Alt + L в IDEA для автоматического выравнивания всего и вся

Diagorn commented on ru/turing/courses/lesson5/Student.java:
```diff
@@ -0,0 +1,158 @@
+     */
+    public static void bubbleSort(String[] array){
+        if(array==null)
+            throw new NullArrayException();
```
Сюда бы текст засунуть какой-то, кидать эксепшны без поясняющих сообщений не самый хороший тон

Diagorn commented on ru/turing/courses/lesson5/Student.java:
```diff
@@ -0,0 +1,158 @@
+     * @param first
+     * @param second
+     */
+    private static void toSwap( String[] array, int first, int second){
```
```suggestion
    private static void swap( String[] array, int first, int second){
```
Так по неймингу корректнее будет

Diagorn commented on ru/turing/courses/lesson5/Student.java:
```diff
@@ -0,0 +1,158 @@
+/**
+ * Класс Student с тремя полями: id, name, phone
+ */
+public class Student {
```
```suggestion
public class Student implements Comparable {
```
Если ты хочешь сделать прям красиво, я бы ещё вставил какую-нибудь сортировку по умолчанию через этот интерфейс

Diagorn commented on ru/turing/courses/lesson5/Student.java:
```diff
@@ -0,0 +1,158 @@
+     * сортировка пузырьком элементов в массиве String[]
+     * @param array
+     */
+    public static void bubbleSort(String[] array){
```
Все сортировки идут в отдельный класс. Задача класса Student - описывать, как хранятся в нём данные. Сортировка - тем более строк и целых чисел - это не его зона ответственности

Diagorn commented on ru/turing/courses/lesson5/Main.java:
```diff
@@ -1,7 +1,67 @@
+
+        List<Student> students = new ArrayList<Student>();
+        System.out.println("Исходные данные");
+        students.add(new Student(Long.valueOf(6), "Olga", "8-999-111-22-33"));
```
`Long.valueOf(6)` можно спокойно заменить на `6L`

TnWqUe committed:
Выполнила работу над ошибками:
1. Ctrl + Alt + L в IDEA для автоматического выравнивания текста
2. Теперь исключение содержит сообщение
3. Исправила название статического метода private static void swap( String[] array, int first, int second)
4. Вставила сортировку по дате поступления студента (по умолчанию)
5. Добавила поле private int yearOfAdmission (дата поступления студента)
6. Создала класс Sorting. Его задача включать статические методы сортировок массивов разных типов данных
7. Заменила Long.valueOf(6)  на 6L

Diagorn commented on ru/turing/courses/lesson5/Sorting.java:
```diff
@@ -0,0 +1,72 @@
+    /**
+     * перестановка 2-х элементов с индексами first, second массива String[]
+     *
+     * @param array
+     * @param first
+     * @param second
```
В JavaDoc `@param`, `@return`, `@see` и другие используются для пояснений, в пустых таких строчках нет смысла. В данном случае я бы написал документацию так:

/**
     * Перестановка элементов в массиве строк
     *
     * @param array - массив, в котором выполняется перестановка
     * @param first - индекс первого переставляемого элемента
     * @param second - индекс второго переставляемого элемента

Diagorn reviewed (approved):
Молодец!
